src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector_SequencePoints.cs

```diff
@@ -170,7 +157,7 @@ internal static void GetBreakpointSpan(VariableDeclaratorSyntax declaratorSyntax
 
         internal static void GetFirstLocalOrFieldBreakpointSpan(SyntaxToken? firstToken, VariableDeclaratorSyntax declaratorSyntax, out SyntaxNode node, out TextSpan? part)
         {
-            var declarationSyntax = (VariableDeclarationSyntax)declaratorSyntax.Parent;
+            var declarationSyntax = (VariableDeclarationSyntax)declaratorSyntax.Parent!;
```

Comment:
>declaratorSyntax.Parent! [](http://example.com/codeflow?start=63&length=24)

Consider adding `Debug.Assert(declaratorSyntax.Parent is { });` instead.

Reply:
I can do that here if you prefer. In general, though, in code that heavily uses `Parent` property, adding all these asserts seems to create a lot of noise.

Reply:
For example: https://github.com/dotnet/roslyn/blob/683e56bdcec7bed3fd2565a7eb26911fb251a134/src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs#L2851-L2859

Reply:
Wondering if we could use an extension method `Parent()` that would return non-null parent (and assert it's not null).